smartsim/_core/generation/operations.py

```diff
@@ -0,0 +1,298 @@
+        return Command(
+            [
+                sys.executable,
+                "-m",
+                entry_point_path,
+                copy_cmd,
+                str(self.src),
+                final_dest,
+                "--dirs_exist_ok",
+            ]
+        )
+
+
+class SymlinkOperation(GenerationProtocol):
+    """Symlink Operation"""
+
+    def __init__(
+        self, src: pathlib.Path, dest: t.Optional[pathlib.Path] = None
+    ) -> None:
+        """Initialize a SymlinkOperation object
+
+        :param src: Path to source
+        :param dest: Path to destination
+        """
+        check_src_and_dest_path(src, dest)
+        self.src = src
+        self.dest = dest
+
+    def format(self, context: GenerationContext) -> Command:
+        """Create Command to invoke symlink file system entry point
+
+        :param context: Context for symlink operation
+        :return: Symlink Command
+        """
+        normalized_path = os.path.normpath(self.src)
+        parent_dir = os.path.basename(normalized_path)
+        final_dest = create_final_dest(context.job_root_path, self.dest)
+        new_dest = os.path.join(final_dest, parent_dir)
+        return Command(
+            [
+                sys.executable,
+                "-m",
+                entry_point_path,
+                symlink_cmd,
+                str(self.src),
+                new_dest,
+            ]
+        )
+
+
+class ConfigureOperation(GenerationProtocol):
+    """Configure Operation"""
+
+    def __init__(
+        self,
+        src: pathlib.Path,
+        file_parameters: t.Mapping[str, str],
+        dest: t.Optional[pathlib.Path] = None,
+        tag: t.Optional[str] = None,
+    ) -> None:
+        """Initialize a ConfigureOperation
+
+        :param src: Path to source
+        :param file_parameters: File parameters to find and replace
+        :param dest: Path to destination
+        :param tag: Tag to use for find and replacement
+        """
+        check_src_and_dest_path(src, dest)
+        self.src = src
+        self.dest = dest
+        pickled_dict = pickle.dumps(file_parameters)
+        encoded_dict = base64.b64encode(pickled_dict).decode("ascii")
+        self.file_parameters = encoded_dict
+        self.tag = tag if tag else ";"
+
+    def format(self, context: GenerationContext) -> Command:
+        """Create Command to invoke configure file system entry point
+
+        :param context: Context for configure operation
+        :return: Configure Command
+        """
+        final_dest = create_final_dest(context.job_root_path, self.dest)
+        return Command(
+            [
+                sys.executable,
+                "-m",
+                entry_point_path,
+                configure_cmd,
+                str(self.src),
+                final_dest,
+                self.tag,
+                self.file_parameters,
+            ]
+        )
+
+
+T = t.TypeVar("T", bound=GenerationProtocol)
```

Comment:
Nit: Usually if I am making a type var, and I am binding it to something, I like to try and give the variable a slightly more descriptive name just so that it is clear what it is bound to. Thoughts on renaming to
```suggestion
GenerationProtocolT = t.TypeVar("GenerationProtocolT", bound=GenerationProtocol)

```
?

Reply:
Ah yes that is super mega reasonable